assets/js/components/adminbar/AdminBarSessionsGA4.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * AdminBarSessionsGA4 component.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import DataBlock from '../DataBlock';
+import PreviewBlock from '../PreviewBlock';
+import { NOTICE_STYLE } from '../GatheringDataNotice';
+import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
+import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import { DATE_RANGE_OFFSET } from '../../modules/analytics/datastore/constants';
+import { calculateChange } from '../../util';
+import { MODULES_ANALYTICS_4 } from '../../modules/analytics-4/datastore/constants';
+const { useSelect } = Data;
+
+const AdminBarSessionsGA4 = ( { WidgetReportError } ) => {
+	const isGatheringData = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).isGatheringData()
+	);
+	const url = useSelect( ( select ) =>
+		select( CORE_SITE ).getCurrentEntityURL()
+	);
+	const dateRangeDates = useSelect( ( select ) =>
+		select( CORE_USER ).getDateRangeDates( {
+			compare: true,
+			offsetDays: DATE_RANGE_OFFSET,
+		} )
+	);
+	const reportArgs = useMemo(
```

Comment:
Yeah, I have thought about why we don't use `useMemo()`. I'm using it to memoize the `reportArgs` and `gatheringDataProps` objects to avoid recomputing them on every re-render.

Reply:
Hmm, this strikes me as a case of premature optimisation. If we don't actually have a performance issue that we're using it to solve, then we should not be adding unneeded complexity (plus memory usage etc) to our code.

I wanted to refresh my own understanding about this stuff quite recently and found this article quite useful: https://www.developerway.com/posts/how-to-use-memo-use-callback. Note that I think we're already using `useCallback()` in a lot of places where we don't actually need to...